### PR TITLE
Transfer learning examples

### PR DESCRIPTION
**Pull request by eleonoreBirgy**
**PR description**
This PR showcases a transfer-learning example using pytorch with mednist dataset .
Provides an example  in notebook and on the fedbiomed website

**Developer Certificate Of Origin (DCO)**

By opening this merge request, you agree to the
[Developer Certificate of Origin (DCO)](https://github.com/fedbiomed/fedbiomed/blob/master/CONTRIBUTING.md#fed-biomed-developer-certificate-of-origin-dco)

This DCO essentially means that:

- you offer the changes under the same license agreement as the project, and
- you have the right to do that,
- you did not steal somebody else’s work.

**License**

Project code files should begin with these comment lines to help trace their origin:
```
# This file is originally part of Fed-BioMed
# SPDX-License-Identifier: Apache-2.0
```

Code files can be reused from another project with a compatible non-contaminating license.
They shall retain the original license and copyright mentions.
The `CREDIT.md` file and `credit/` directory shall be completed and updated accordingly.

**Guidelines for PR review**

General:

* give a glance to [DoD](https://fedbiomed.org/latest/developer/definition-of-done/)
* check [coding rules and coding style](https://fedbiomed.org/latest/developer/usage_and_tools/#coding-style)
* check docstrings (eg run `tests/docstrings/check_docstrings`)

Specific to some cases:

* update all conda envs consistently (`development` and `vpn`, Linux and MacOS)
* if modified researcher (eg new attributes in classes) check if breakpoint needs update (`breakpoint`/`load_breakpoint` in `Experiment()`, `save_state_breakpoint`/`load_state_breakpoint` in aggregators, strategies, secagg, etc.)
* if modified a component with versioning (config files, breakpoint, messaging protocol) then update the version following the rules in `common/utils/_versions.py`

### CODE REVIEW

File: docs/tutorials/pytorch/06_Transfer-learning_tutorial_usingDenseNet-121.ipynb [hunk 1]
```diff
@@ -0,0 +1,812 @@
...
+    "# Transfer-learning tutorial using DenseNet-121 pre-trained model: example on MedNIST dataset"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6d343c58-9d11-400f-9e2e-9b0d1e840105",
+   "metadata": {},
+   "source": [
+    "\n",
+    "## Goal of this tutoriel\n",
```
Comment (luciainnocenti) [inline docs/tutorials/pytorch/06_Transfer-learning_tutorial_usingDenseNet-121.ipynb:17]: Typo: tutorial

File: docs/tutorials/pytorch/06_Transfer-learning_tutorial_usingDenseNet-121.ipynb [hunk 2]
```diff
@@ -0,0 +1,812 @@
...
+    "    └── CXR/\n",
+    "\n",
+    "    └── ChestCT/\n",
+    "\n",
+    "    └── Hand/\n",
+    "\n",
+    "    └── HeadCT/   \n",
+    "   \n",
+    "\n",
+    "## Transfer-learning\n",
```
Comment (luciainnocenti) [inline docs/tutorials/pytorch/06_Transfer-learning_tutorial_usingDenseNet-121.ipynb:64]: I would add a disclaimer that we are not doing federated transfer learning

**Review by luciainnocenti — COMMENTED** [review]
Hi everyone, and thanks a lot for adding this important feature in fedbiomed. I like the way it is done and used. I have a minor suggestions about the notebook: what if you show first a training without the pre-trained model, plotting the loss, and after using the pre-trained model in the two configurations (partial and last-layer unfreezing) with again a plot of the losses? I think it would highlight that the loading phase actually succeeded and everything works as expected.

**Comment by eleonoreBirgy** [comment]
I agree Lucia. I also thought about doing an experiment on an untrained model, in order to illustrate the interest of transfer learning and give like a proof of concept. I preferred to illustrate an example of partial fine tuning where we partially train the model. To my mind you could choose to train the entire model when you have a really really  big dataset. But illustrate both strategies (training without pretrained model, and training with pretrained model with the two illustrated ways: partial and last-layer unfreezing) and compare evauation metrics is a good and complete scientific demonstration. Let me know if you want to discuss more about this topic....

File: docs/tutorials/pytorch/06_Transfer-learning_tutorial_usingDenseNet-121.ipynb [hunk 3]
```diff
@@ -0,0 +1,812 @@
...
+   "execution_count": null,
+   "id": "4035e2a7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model = torch.hub.load('pytorch/vision:v0.10.0', 'densenet121', pretrained=True)\n",
+    "torch.save(model.state_dict(), 'pretrained_model.pt')"
+   ]
+  },
+  {
```
[the remark this answers is not in the document]
Comment (eleonoreBirgy) [reply docs/tutorials/pytorch/06_Transfer-learning_tutorial_usingDenseNet-121.ipynb:265]: yes

File: docs/tutorials/pytorch/06_Transfer-learning_tutorial_usingDenseNet-121.ipynb (hunk 2 above)
Comment (ybouilla) [reply docs/tutorials/pytorch/06_Transfer-learning_tutorial_usingDenseNet-121.ipynb:64]: Is FTL what is referred on this webpage? https://developers.sherpa.ai/tutorials/federated-learning-paradigms/ftl

Comment (luciainnocenti) [reply docs/tutorials/pytorch/06_Transfer-learning_tutorial_usingDenseNet-121.ipynb:64]: Yes, it is.

**Comment by ybouilla** [comment]
Hi everyone, I have more updates from the team:
- we may want to make  shorter the notebook
- we may pass the number of layers to unfrozen in the `model_args` argument
- transfer learning notebook should be in the transfer learning notebook folder
- show with mednist image how well the classifier works
- display the model weights before/after experiment and see which layers have changed /unchanged during training

**Comment by ybouilla** [comment]
Hi, I have just removed some notebooks and moved the transfer learning notebook inside `notebook/transfer-learning` folder
